docs/scos/dev/front-end-development/202212.0/oryx/reactivity/signals.md

```diff
@@ -0,0 +1,118 @@
+
+Here's an example of using options:
+
+```ts
+const values = signal(observable$, { initialValue: 1, equal: (a, b) => a === b });
+```
+
+### Computed Signals
+
+A computed signal derives its value from other signals. It automatically reevaluates its value when any of the signals it depends on changes.
+
+Here's an example of a computed signal:
+
+```ts
+const counter = computed(() => 3 * counter(1));
+```
+
+Computed signals can also convert observables to signals transparently:
+
+```ts
+const counter = computed(() => productService.get({ sku: productSku() }));
+```
+
+In the example above, `productSku` is a signal, and `productService.get` returns an observable.
+
+Computed signals can use the same set of options as regular signals.
+
+### Effects
+
+Effects are functions that run whenever a signal's value changes.
+
+Here's an example:
+
+```ts
+const counter = effect(() => {
+    console.log('counter changed', counter());
+});
+```
+
+You can configure effects using options to modify their behavior:
+
+- `defer`: If this is set to true, your effect won't run until you explicitly call the `start()` method.
+- `async`: Set this to true if you want your effect to run asynchronously.
+
+Example:
+
+```ts
+const counter = effect(() => {
+    console.log('counter changed', counter());
+}, { defer: true, async: true });
+```
+
+## Using Signals in Components
+
+### @signalAware directive
+
+The `@signalAware` decorator provides additional functionality when using signals in components.
+
+```ts
+@signalAware()
+class MyComponent extends LitElement {}
+```
+
+This decorator is required to make your component work with signals as expected.
+When used, the component will automatically detect signals and render changes whenever a signal alters. It does this intelligently, considering only the signals relevant to the last render.
+
+Some Oryx domain components are not using this decorator directly, as it is already applied to some common domain mixins (eg. `ContentMixin`, `ProductMixin`, etc.).
+
+### `@elementEffect` Directive
+
+The `@elementEffect` directive integrates effects with component lifecycles for seamless management. It activates an effect when a component is connected to the DOM and deactivates it once the component is disconnected.
+
+```ts
+class MyComponent extends LitElement {
+  /* ... */  
+  @elementEffect()
+  logProductCode = () => console.log('Product code ', this.$product().code);
+}
+```
+
+In the above example, the `logProductCode` effect will automatically start when `MyComponent` is connected to the DOM, logging the product code everytime $product signal will update. Once the component is disconnected from the DOM, the effect will automatically stop. This directive simplifies effect management by automatically linking them to component lifecycles, making your component code cleaner and easier to manage.
```

Comment:
In general, avoid future tenses as much as possible. Also, this sentence looks unclear, so consider rephrasing or splitting it into two smaller ones: "In the preceding example, the `logProductCode` effect automatically starts when `MyComponent` is connected to the DOM, logging the product code every time, and the $product signal updates."
```suggestion
In the above example, the `logProductCode` effect starts automatically when `MyComponent` is connected to the DOM, logging the product code every time the $product signal updates. Once the component is disconnected from the DOM, the effect automatically stops. This directive simplifies effect management by automatically linking them to component lifecycles, making your component code cleaner and easier to manage.
```

Reply:
improved